### PR TITLE
Fix irc.py and py2

### PR DESCRIPTION
**Pull request by godwhoa**
Changes:
Upgrade to new command system
Replace itertools.zip with six.move.zip

Replace FileExistsError with errno.EEXIST for py2 compat.
Fix kwarg in irc_say
Fix condif_dir typo in save_bans
Make filter_printable less fancy for py3 support
Fix nick encoding issue with IRC client
Fix attempted nick errors by adding nickname setter

Sorry for clumping all those changes into one commit

### CODE REVIEW

File: piqueserver/server.py [hunk 1]
```diff
@@ -95,8 +96,11 @@ def ensure_dir_exists(filename):
     d = os.path.dirname(filename)
     try:
         os.makedirs(d)
-    except FileExistsError:
-        pass
+    except OSError as e:
+        if e.errno == errno.EEXIST:
+            pass
+        else:
+            raise
```
Comment (swalladge) [inline piqueserver/server.py]: should be `raise e`

Comment (godwhoa) [reply piqueserver/server.py]: @swalladge fixed the issues in latest commit can you re-review it?

Comment (swalladge) [reply piqueserver/server.py]: @godwhoa looks good to me. 👍

**Comment by dgadelha** [comment]
@swalladge the piqueserver console might need an update then:
image

**Comment by godwhoa** [comment]
@dgadelha  That one comes from [piqueserver/player.py#L350](https://github.com/piqueserver/piqueserver/blob/master/piqueserver/player.py#L349-L350). It'll get fixed in a future PR for [improving logging](https://github.com/piqueserver/piqueserver/issues?utf8=%E2%9C%93&q=is%3Aissue+is%3Aopen+logging).

**Comment by NotAFile** [comment]
That's really easy to fix actually... in fact, iirc, I commited it before

**Pull request merged**